cmd/otel-allocator/main.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/fsnotify/fsnotify"
+	gokitlog "github.com/go-kit/log"
+	"github.com/gorilla/mux"
+
+	"github.com/otel-allocator/allocation"
+
+	"github.com/otel-allocator/collector"
+	"github.com/otel-allocator/config"
+	lbdiscovery "github.com/otel-allocator/discovery"
+)
+
+const (
+	configDir  = "/conf/"
+	listenAddr = ":8080"
+)
+
+func main() {
+	ctx := context.Background()
+
+	// watcher to monitor file changes in ConfigMap
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Fatalf("Can't start the watcher: %v", err)
+	}
+	defer watcher.Close()
+
+	if err := watcher.Add(configDir); err != nil {
+		log.Fatalf("Can't add directory to watcher: %v", err)
+	}
+
+	srv, err := newServer(listenAddr)
+	if err != nil {
+		log.Fatalf("Can't start the server: %v", err)
+	}
+
+	interrupts := make(chan os.Signal, 1)
+	signal.Notify(interrupts, os.Interrupt, syscall.SIGTERM)
+
+	go func() {
+		if err := srv.Start(); err != http.ErrServerClosed {
+			log.Fatalf("Can't start the server: %v", err)
+		}
+	}()
+
+	for {
+		select {
+		case <-interrupts:
+			if err := srv.Shutdown(ctx); err != nil {
+				log.Println(err)
+				os.Exit(1)
+			}
+			os.Exit(0)
+		case event := <-watcher.Events:
+			switch event.Op {
+			case fsnotify.Create:
+				log.Println("ConfigMap updated!")
```

Comment:
Have you tested this logic in a container deployed on Kubernetes?

Reply:
Yes this has been tested on a minikube cluster locally.